### ISSUE TITLE
Relicense under dual MIT/Apache-2.0

### ISSUE THREAD

**Issue by cmr** [rust-lang-nursery/lazy-static.rs#30]
Someone explicitly asked for this issue to be opened.

TL;DR the Rust ecosystem is largely Apache-2.0. Being available under that
license is good for interoperation. The MIT license as an add-on can be nice
for GPLv2 projects to use your code.
# Why?

The MIT license requires reproducing countless copies of the same copyright
header with different names in the copyright field, for every MIT library in
use. The Apache license does not have this drawback. However, this is not the
primary motivation for me creating these issues. The Apache license also has
protections from patent trolls and an explicit contribution licensing clause.
However, the Apache license is incompatible with GPLv2. This is why Rust is
dual-licensed as MIT/Apache (the "primary" license being Apache, MIT only for
GPLv2 compat), and doing so would be wise for this project. This also makes
this crate suitable for inclusion and unrestricted sharing in the Rust
standard distribution and other projects using dual MIT/Apache, such as my
personal ulterior motive, [the Robigalia project](https://robigalia.org).

Some ask, "Does this really apply to binary redistributions? Does MIT really
require reproducing the whole thing?" I'm not a lawyer, and I can't give legal
advice, but some Google Android apps include open source attributions using
this interpretation. [Others also agree with
it](https://www.quora.com/Does-the-MIT-license-require-attribution-in-a-binary-only-distribution).
But, again, the copyright notice redistribution is not the primary motivation
for the dual-licensing. It's stronger protections to licensees and better
interoperation with the wider Rust ecosystem.
# How?

To do this, get explicit approval from each contributor of copyrightable work
(as not all contributions qualify for copyright, due to not being a "creative
work", e.g. a typo fix) and then add the following to your README:

```
## License

Licensed under either of

 * Apache License, Version 2.0, ([LICENSE-APACHE](LICENSE-APACHE) or http://www.apache.org/licenses/LICENSE-2.0)
 * MIT license ([LICENSE-MIT](LICENSE-MIT) or http://opensource.org/licenses/MIT)

at your option.

### Contribution

Unless you explicitly state otherwise, any contribution intentionally submitted
for inclusion in the work by you, as defined in the Apache-2.0 license, shall be dual licensed as above, without any
additional terms or conditions.
```

and in your license headers, if you have them, use the following boilerplate
(based on that used in Rust):

```
// Copyright 2016 lazy-static.rs Developers
//
// Licensed under the Apache License, Version 2.0, <LICENSE-APACHE or
// http://apache.org/licenses/LICENSE-2.0> or the MIT license <LICENSE-MIT or
// http://opensource.org/licenses/MIT>, at your option. This file may not be
// copied, modified, or distributed except according to those terms.
```

It's commonly asked whether license headers are required. I'm not comfortable
making an official recommendation either way, but the Apache license
recommends it in their appendix on how to use the license.

Be sure to add the relevant `LICENSE-{MIT,APACHE}` files. You can copy these
from the [Rust](https://github.com/rust-lang/rust) repo for a plain-text
version.

And don't forget to update the `license` metadata in your `Cargo.toml` to:

```
license = "MIT OR Apache-2.0"
```

I'll be going through projects which agree to be relicensed and have approval
by the necessary contributors and doing this changes, so feel free to leave
the heavy lifting to me!
# Contributor checkoff

To agree to relicensing, comment with :

```
I license past and future contributions under the dual MIT/Apache-2.0 license, allowing licensees to chose either at their option.
```

Or, if you're a contributor, you can check the box in this repo next to your
name. My scripts will pick this exact phrase up and check your checkbox, but
I'll come through and manually review this issue later as well.
- [x] @Kimundi
- [ ] @drewcrawford
- [x] @SimonSapin
- [x] @cmr
- [x] @tov
- [x] @niax
- [x] @frewsxcv
- [x] @durka
- [x] @HeroesGrave
- [x] @badboy
- [x] @Ericson2314
- [x] @ktossell
- [x] @malbarbo
- [x] @Ryman
- [x] @mamartel

**Comment by frewsxcv** [comment]
I license past and future contributions under the dual MIT/Apache-2.0 license, allowing licensees to chose either at their option.

**Comment by mamartel** [comment]
I license past and future contributions under the dual MIT/Apache-2.0 license, allowing licensees to chose either at their option.

**Comment by SimonSapin** [comment]
I license past and future contributions under the dual MIT/Apache-2.0 license, allowing licensees to chose either at their option.

**Comment by durka** [comment]
I license past and future contributions under the dual MIT/Apache-2.0
license, allowing licensees to chose either at their option.

**Comment by HeroesGrave** [comment]
I license past and future contributions under the dual MIT/Apache-2.0 license, allowing licensees to chose either at their option.

**Comment by jadpole** [comment]
I license past and future contributions under the dual MIT/Apache-2.0 license, allowing licensees to chose either at their option.

**Comment by tov** [comment]
I license past and future contributions under the dual MIT/Apache-2.0 license, allowing licensees to chose either at their option.

**Comment by badboy** [comment]
I license past and future contributions under the dual MIT/Apache-2.0 license, allowing licensees to chose either at their option.

**Comment by Ryman** [comment]
I license past and future contributions under the dual MIT/Apache-2.0 license, allowing licensees to chose either at their option.

**Comment by Kimundi** [comment]
I license past and future contributions under the dual MIT/Apache-2.0 license, allowing licensees to chose either at their option.

**Comment by Ericson2314** [comment]
I license past and future contributions under the dual MIT/Apache-2.0 license, allowing licensees to chose either at their option.

**Comment by dtolnay** [comment]
@drewcrawford @niax @ktossell @malbarbo would you be okay with licensing your past and future lazy-static contributions under the dual MIT/Apache-2.0 license, allowing licensees to choose either at their option?

**Comment by niax** [comment]
I license past and future contributions under the dual MIT/Apache-2.0 license, allowing licensees to chose either at their option.

**Comment by ktossell** [comment]
This is fine with me, too.

On April 4, 2017 8:18:10 PM EDT, Nicholas Hollett <notifications@github.com> wrote:
>I license past and future contributions under the dual MIT/Apache-2.0
>license, allowing licensees to chose either at their option.
>
>-- 
>You are receiving this because you were mentioned.
>Reply to this email directly or view it on GitHub:
>https://github.com/rust-lang-nursery/lazy-static.rs/issues/30#issuecomment-291686600

-- 
Sent from my Android device with K-9 Mail. Please excuse my brevity.

**Comment by malbarbo** [comment]
I license past and future contributions under the dual MIT/Apache-2.0 license, allowing licensees to chose either at their option.

**Comment by vishvananda** [comment]
It looks like this has been open for 1.5 years. Is it possible to track down the last few people and push this change through?

**Comment by dtolnay** [comment]
Looks like none of [@drewcrawford's contributions](https://github.com/rust-lang-nursery/lazy-static.rs/commits?author=drewcrawford) are copyrightable work so I believe that covers all of the contributors as of Jan 22, 2016. There have been three new contributors since then, each with a single commit:

- [ ] @astraw https://github.com/rust-lang-nursery/lazy-static.rs/commits?author=astraw
- [ ] @emilio https://github.com/rust-lang-nursery/lazy-static.rs/commits?author=emilio
- [ ] @GuillaumeGomez https://github.com/rust-lang-nursery/lazy-static.rs/commits?author=GuillaumeGomez

These are also not copyrightable work so @cmr @Kimundi we're all clear here.

**Comment by GuillaumeGomez** [comment]
Ok for me.

**Comment by astraw** [comment]
I license past and future contributions under the dual MIT/Apache-2.0 license, allowing licensees to chose either at their option.

**Comment by Kimundi** [comment]
Alright, then I think we only need @neosilky's permission for the the few PRs I just merged. :)

**Comment by neosilky** [comment]
I license past and future contributions under the dual MIT/Apache-2.0 license, allowing licensees to chose either at their option.

**Comment by Kimundi** [comment]
Great! Making a PR now...

**Comment by Kimundi** [comment]
I just released a new version with the dual-licensing applied.

If anyone could double check that I haven't forgotten anything that would be nice. :)

**Comment by neosilky** [comment]
You bumped the version so you also need to increment https://github.com/rust-lang-nursery/lazy-static.rs/blob/master/src/lib.rs#L95 :grinning:

**Comment by Kimundi** [comment]
Dangit! xD

**Comment by neosilky** [comment]
Maybe https://github.com/rust-lang-nursery/lazy-static.rs/blob/master/Cargo.toml#L3 should have been in capitals :stuck_out_tongue_closed_eyes:

**Comment by neosilky** [comment]
Whilst we're on the topic of versioning, I just noticed the README contains

```toml
[dependencies]
lazy_static = "0.2"
```

**Comment by Kimundi** [comment]
@neosilky: Yeah, my bad. My eyes just jumped directly to version field, even though I explicitly acknowledge that comment during merging earlier. :P I fixed it just now.

The `0.2` In the README is fine, its interpreted by cargo as `^0.2` according to Semver, which means the highest known `0.2.x` matches it.

**Comment by neosilky** [comment]
@Kimundi Ah I see, good job!

**Comment by Kimundi** [comment]
I think this can be closed now
